### ISSUE TITLE
SSE4.2 functions

### ISSUE THREAD

**Issue by nemequ** [nemequ/simde#7]
Per https://software.intel.com/sites/landingpage/IntrinsicsGuide/#techs=SSE4_2

 * [ ] `_mm_cmpestra`
 * [ ] `_mm_cmpestrc`
 * [ ] `_mm_cmpestri`
 * [ ] `_mm_cmpestrm`
 * [ ] `_mm_cmpestro`
 * [ ] `_mm_cmpestrs`
 * [ ] `_mm_cmpestrz`
 * [ ] `_mm_cmpgt_epi64`
 * [ ] `_mm_cmpistra`
 * [ ] `_mm_cmpistrc`
 * [ ] `_mm_cmpistri`
 * [ ] `_mm_cmpistrm`
 * [ ] `_mm_cmpistro`
 * [ ] `_mm_cmpistrs`
 * [ ] `_mm_cmpistrz`

**Comment by mr-c** [comment]
@nemequ Could you set up the build / test infrastructure for this? There is some interest @ FU Berlin in using SIMDe w/ SSE4.2

**Comment by nemequ** [comment]
It's all ready to go, though it was a bit difficult to understand how to use it without looking at another ISA extension.  I just added `_mm_cmpgt_epi64` (11b0c2e7e08afbddad99ebb26ed66b52568a8ab7) as an example, hopefully that helps clear things up.

**Comment by FaresSalem** [comment]
I might consider this issue to work on as part of GSoC, is anybody working on it already?

**Comment by nemequ** [comment]
Yes, @masterchef2209 is planning to work on this.  Sorry.

**Comment by pthariensflame** [comment]
Any progress on this?

**Comment by jserv** [comment]
> I'm missing intrinsic `_mm_cmpestri`.

Recently, [SSE2NEON](https://github.com/DLTcollab/sse2neon) implements both `_mm_cmpestri` and `_mm_cmpestrm`.

**Comment by mr-c** [comment]
> > I'm missing intrinsic `_mm_cmpestri`.
> 
> Recently, [SSE2NEON](https://github.com/DLTcollab/sse2neon) implements both `_mm_cmpestri` and `_mm_cmpestrm`.

Yes, they have all of the SSE4.2 functions we are missing; porting those to SIMDe style would be greatly appreciated!

https://github.com/DLTcollab/sse2neon/blob/39d854002840f12baa85cfdf01b120eabdfa76ed/sse2neon.h#L8093-L8097
